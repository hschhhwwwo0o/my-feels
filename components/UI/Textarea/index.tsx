import React, { Dispatch, FunctionComponent, SetStateAction } from "react";

interface ITextarea {
  value: string;
  setValue?: Dispatch<SetStateAction<string>>;
}

const Textarea: FunctionComponent<ITextarea> = ({ value = "", setValue = () => {} }) => {
  return (
    <textarea
      value={value}
      onChange={e => {
        setValue(e.target.value);
      }}
      className="mt-5 w-full bg-transparent font-medium text-[#444444] dark:text-[#B2B2B2] outline-none text-justify h-[3000px] placeholder-[#bebebe] dark:placeholder-[#545454]"
      placeholder="Enter text"
    />
  );
};

export default Textarea;
