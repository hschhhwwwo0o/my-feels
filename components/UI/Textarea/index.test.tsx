import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Textarea from "./index";

test("Test Textarea component", () => {
  render(<Textarea value={"text"} setValue={() => ""} />);
  const el: HTMLElement = screen.getByRole("textbox");
  expect(el).toBeInTheDocument();
  expect(el).toMatchSnapshot();
});
