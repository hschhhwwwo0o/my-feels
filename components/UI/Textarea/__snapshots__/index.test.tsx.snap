// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Textarea component 1`] = `
<textarea
  class="mt-5 w-full bg-transparent font-medium text-[#444444] dark:text-[#B2B2B2] outline-none text-justify h-[3000px] placeholder-[#bebebe] dark:placeholder-[#545454]"
  placeholder="Enter text"
  role="textbox"
>
  text
</textarea>
`;
